src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -548,7 +548,8 @@ void program::pre_optimize_graph(bool is_internal) {
         // Ideally this should be done before fusing to simplify logic and make the pass more powerful,
         // but after format selection to select correct alignment.
         // Unfortunately those passes currently happen in reverse order.
-        apply_opt_pass<concat_input_order>();
+        if (!options.get<build_option_type::is_dynamic>()->enabled())
```

Comment:
Why do we need that option? I think the same can be easily achieved by adding node shape check inside concat optimization pass:
```cpp
        if (!node->is_type<concatenation>() || node->is_output() || node->is_dynamic())
            continue;
```

Reply:
Fixed